src/backend/distributed/utils/metadata_cache.c

```diff
@@ -280,6 +302,53 @@ LookupDistTableCacheEntry(Oid relationId)
 		heap_close(distShardRelation, AccessShareLock);
 	}
 
+	/* decide and allocate interval comparison function */
+	if (shardIntervalArrayLength > 0)
+	{
+		MemoryContext oldContext = CurrentMemoryContext;
+
+		/* allocate the comparison function in the cache context */
+		oldContext = MemoryContextSwitchTo(CacheMemoryContext);
+
+		shardIntervalCompareFunction = ShardIntervalCompareFunction(shardIntervalArray,
+																	partitionMethod);
+
+		MemoryContextSwitchTo(oldContext);
+	}
+
+	/* sort the interval array */
+	sortedShardIntervalArray = SortShardIntervalArray(shardIntervalArray,
+													  shardIntervalArrayLength,
+													  shardIntervalCompareFunction);
+
+	/* check the shard distribution for hash partitioned tables */
+	if (partitionMethod == DISTRIBUTE_BY_HASH)
+	{
+		hasUniformHashDistribution =
+			HasUniformHashDistribution(sortedShardIntervalArray,
+									   shardIntervalArrayLength);
+	}
+
+	/* check if there exists any shard intervals with no min/max values */
+	hasUninitializedShardInterval =
+		HasUninitializedShardInterval(sortedShardIntervalArray, shardIntervalArrayLength);
+
+	/* we only need hash functions for hash distributed tables */
+	if (partitionMethod == DISTRIBUTE_BY_HASH)
```

Comment:
Yeah, that makes sense. I collapsed both `DISTRIBUTE_BY_HASH ` into a single if.